### ISSUE TITLE
Spaceline with spaceline-all-the-icons broken in Makefile mode

### ISSUE THREAD

**Issue by brunosmmm** [syl20bnr/spacemacs#13292]
#### Description :octocat:
Spaceline rendering is broken in Makefile mode, many times taller than normal

#### Reproduction guide :beetle:
- Start Emacs
- Change spaceline theme to ```all-the-icons```
- Open a Makefile

```emacs-lisp
(defun dotspacemacs/user-init ()
  (setq dotspacemacs-mode-line-theme '(all-the-icons)))
```

*Observed behaviour:* :eyes: :broken_heart:
spaceline is much taller than it should be

[![makefile.png](https://i.postimg.cc/5N3BC1vD/makefile.png)](https://postimg.cc/1f8nkkvK)

Note: currently using 27.0.60 but I have observed the exact same with 26.3 for quite a while

*Expected behaviour:* :heart: :smile:
spaceline has proper height as in every other mode

#### System Info :computer:
- OS: gnu/linux
- Emacs: 27.0.60
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 0a06c7ca6)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(emacs-lisp helm multiple-cursors treemacs)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND GPM DBUS GSETTINGS GLIB NOTIFY INOTIFY ACL GNUTLS LIBXML2 FREETYPE HARFBUZZ M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS LIBSYSTEMD JSON PDUMPER LCMS2 GMP

**Comment by duianto** [comment]
I'm guessing that your using a HiDPi screen.

Maybe it can be fixed by setting the mode line theme `separator-scale`:
https://github.com/syl20bnr/spacemacs/blob/develop/doc/DOCUMENTATION.org#separator-scale

I don't have a HiDPi screen and the mode line looks fine for me when I open a makefile, for example:
`c:/Users/username/.emacs.d/.cache/quelpa/build/cpp-auto-include/Makefile`

<details><summary>
PopOS 19.10
</summary>
<pre>
#### System Info :computer:
- OS: gnu/linux
- Emacs: 26.3
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. f03392573)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(auto-completion command-log emacs-lisp git helm markdown multiple-cursors
                 (org :variables org-agenda-files
                      '("~/org/notes.org"))
                 (shell :variables shell-default-shell 'eshell shell-default-height 30 shell-default-position 'bottom)
                 spell-checking syntax-checking treemacs version-control)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS GLIB NOTIFY ACL LIBSELINUX GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM THREADS LIBSYSTEMD LCMS2
</pre>
</details>

![VirtualBoxVM_2020-02-21_19-35-43](https://user-images.githubusercontent.com/13420573/75061808-c6542700-54e1-11ea-83c5-e12a7773bdac.png)

---

<details><summary>
Windows 1903
</summary>
<pre>
#### System Info :computer:
- OS: windows-nt
- Emacs: 26.3
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. f03392573)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(autohotkey auto-completion command-log emacs-lisp git github helm imenu-list javascript
            (javascript :variables javascript-backend 'lsp javascript-lsp-linter nil node-add-modules-path t)
            (markdown :variables markdown-live-preview-engine 'vmd markdown-command "vmd")
            multiple-cursors org react
            (shell :variables shell-default-shell 'eshell shell-default-height 30 shell-default-position 'bottom)
            spell-checking syntax-checking treemacs version-control)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS THREADS LCMS2
</pre>
</details>

![emacs_2020-02-21_19-25-57](https://user-images.githubusercontent.com/13420573/75061812-c94f1780-54e1-11ea-9223-5d0ac5cb179c.png)

**Comment by brunosmmm** [comment]
@duianto I'm not using a HiDPi screen and separator-scale doesn't do anything !

**Comment by smile13241324** [comment]
I am afraid I cannot reproduce this issue. I am on Windows running Spacemacs in a docker container and transfering the screen via sshx. I have just made a new makefile and changed the spacemacs modeline theme to `all-the-icons`. Apart from me having to increase the separator scale from 1.2 to 2.0 I had not to change anything else. The modeline looks ok, apart from a small graphical glich I see on the left side of the clock which looks like it is caused by one of the separators.

Anyway, can you both please make sure you have the latest version of the fonts installed by calling `all-the-icons-install-fonts` and update your branches and packages.

In addition @brunosmmm do you have any error messages in the message buffer right after starting spacemacs? I am looking for a sign that your config is not fully loaded. If this is still not working please deactivate any user-init or user-config code in your dotfile and try again.

If it still does not help I would like to ask for your dotfile.

Btw: You said that changing the separator scale does not do anything, does that mean it didn't solve your issue or has the setting had no effect on your mode line at all?

**Comment by brunosmmm** [comment]
I made the report with a freshly cloned spacemacs repo from the develop branch, untouched .spacemacs. Separator scale has no effect, I've had it at 1.5 and 1.0 in the past and see no difference. I have discovered that in another machine that is available to me, I don't get this Makefile modeline weirdness. I'll reproduce again and look for any other signs of anything that might be going on, but if I can't find it I'll come back and close the issue in a few days.
